Use `onCleanup` to close all files

egeerardyn opened the pull request:
Some `try-catch` constructions -- that rethrew the error AFTER `fclose(fid)` --
were also removed, since these were no longer needed.

This will close #508 when merged.

PeterPablo commented:
Code looks good and testsuite ran well in octave (and Travis agrees ;-) ). :+1: 

One minor remark: Since code with `onCleanup` is not that common, it might be a good idea to add `% see #508` in all lines with `onCleanup`.

PeterPablo commented:
`grep` found one left over case of `fclose` in `./test/private/VersionControlIdentifier.m`. Would you please adapt there as well or add a short comment, since it is not really needed?

egeerardyn commented:
I've addressed most of your comments.

The only thing I don't like is adding such a comment as that is a standard (albeit somewhat uncommon) MATLAB construct. Looking up #508 from MATLAB takes longer than just typing `help onCleanup`, for people who don't understand it from the context.
Instead, I've renamed the placeholder variable to `finally_fclose_fid` to communicate the intent more clearly and in the same jargon used in almost every other programming language.
